libvast/src/concept/parseable/vast/expression.cpp

```diff
@@ -114,23 +115,43 @@ static expression expand(data x) {
   return caf::visit(expander{}, expression{std::move(pred)});
 }
 
-static auto make_predicate_parser() {
-  using parsers::alnum;
+expression expand_extractor(predicate::operand operand) {
+  return predicate{
+    std::move(operand),
+    relational_operator::not_equal,
+    predicate::operand{data{}},
+  };
+}
+
+auto make_field_char_parser() {
   using parsers::chr;
+  return parsers::alnum | chr{'_'} | chr{'-'} | chr{':'};
```

Comment:
I find it strange that this is out of sync with the `field_name` from the `legacy_type` parser:
```
  auto field_name = parsers::identifier | parsers::qqstr;
```
The identifier parser is defined standalone as:
```
constexpr inline auto identifier_char = (alnum | ch<'_'> | ch<'.'>);
constexpr inline auto identifier = +identifier_char;
```
It would be great to turn this into a hierarchy that builds on shared blocks. Alternatively the relationship needs to be documented in both places.

Reply:
We agreed to resolve this issue later, as part of tenzir/issues#30.